cupyx/scipy/ndimage/measurements.py

```diff
@@ -510,3 +510,493 @@ def standard_deviation(input, labels=None, index=None):
+        labels (cupy.ndarray, optional): An array of integers marking different
+            regions over which the position of the maximum value of `input` is
+             to be computed. `labels` must have the same shape as `input`. If
+             `labels` is not specified, the location of the first maximum over
+             the whole array is returned.
+
+            The `labels` argument only works when `index` is specified.
+        index (array_like, optional): A list of region labels that are taken
+            into account for finding the location of the maxima. If `index` is
+            None, the ``first`` maximum over all elements where `labels` is
+            non-zero is returned.
+
+            The `index` argument only works when `labels` is specified.
+
+    Returns:
+        output (list of tuples of ints): Tuple of ints or list of tuples of
+            ints that specify the location of maxima of `input` over the
+            regions determaxed by `labels` and  whose index is in `index`.
+
+            If `index` or `labels` are not specified, a tuple of ints is
+            returned specifying the location of the first maximal value of
+            `input`.
+
+    .. note::
+        When `input` has multiple identical maxima within a labeled region,
+        the coordinates returned are not guaranteed to match those returned by
+        SciPy.
+
+    .. seealso:: :func:`scipy.ndimage.maximum_position`
+    """
+    dims = numpy.asarray(input.shape)
+    # see numpy.unravel_index to understand this line.
+    dim_prod = numpy.cumprod([1] + list(dims[:0:-1]))[::-1]
+
+    result = _select(input, labels, index, find_max_positions=True)[0]
+
+    # have to transfer result back to the CPU to return index tuples
+    if result.ndim == 0:
+        result = int(result)
+    else:
+        result = cupy.asnumpy(result)
+
+    if cupy.isscalar(result):
+        return tuple((result // dim_prod) % dims)
+
+    return [tuple(v) for v in (result.reshape(-1, 1) // dim_prod) % dims]
+
+
+def extrema(input, labels=None, index=None):
+    """Calculate the minimums and maximums of the values of an array at labels,
+    along with their positions.
+
+    Args:
+        input (cupy.ndarray) N-D image data to process.
+        labels (cupy.ndarray, optional): Labels of features in input. If not
+            None, must be same shape as `input`.
+        index (int or sequence of ints, optional): Labels to include in output.
+            If None (default), all values where non-zero `labels` are used.
+
+    Returns:
+        minimums (cupy.ndarray): Values of minimums in each feature.
+        maximums (cupy.ndarray): Values of maximums in each feature.
+        min_positions (tuple or list of tuples): Each tuple gives the N-D
+            coordinates of the corresponding minimum.
+        max_positions (tuple or list of tuples): Each tuple gives the N-D
+            coordinates of the corresponding maximum.
```

Comment:
It seems that Sphinx (or its extension) strictly allows only a return value with the following grammar:

    Returns:
        <Return type>: <Description>

As the return type is optional and the description can be multiple lines, would you arrange the part as something like below? (I tried to escape the colon, but sphinx did not read the part as so)


```suggestion
    Returns:
        A tuple that contains the following values.

        minimums (cupy.ndarray): Values of minimums in each feature.

        maximums (cupy.ndarray): Values of maximums in each feature.

        min_positions (tuple or list of tuples):
            Each tuple gives the N-D coordinates of the corresponding minimum.

        max_positions (tuple or list of tuples):
            Each tuple gives the N-D coordinates of the corresponding maximum.
```